dynamic_stack_decider_visualization/dynamic_stack_decider_visualization/dsd_visualization_plugin.py

```diff
@@ -236,7 +239,7 @@ def set_dsd(self, name):
         """
         # close debug connection of old dsd
         if self.dsd is not None:
-            self.dsd.close()
+            self.dsd = None
```

Comment:
Why would there be a "double destruction of the subscription"?

Because with close(), the subscription gets destroyed and then with the removal of the old DSDFollower again?

Wouldn't it then be cleaner to also set `self.debug_subscriber = None` in the close method or alternatively remove the close method completely, as it should not be used?

Reply:
But setting the sub to none would be equivalent to dereferencing the whole object

Reply:
Yes, but that would be okay, wouldn't it?
I interpreted the issue being, that

1. with `self.dsd.close()` `self.debug_subscriber.destroy()` gets called (a reference still exists (unless this somehow removes itself from the reference?!?)) and then,
2. the whole `DsdFollower` gets dereferenced when overwriting `self.dsd = dsd`, leading to a `del()` call on the `self.debug_subscriber`object.